graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -110,6 +113,50 @@ def get_edge_schema_element_or_raise(self, edge_classname):
 
         return schema_element
 
+    def get_unique_indexes_for_class(self, cls):
+        """Return a frozenset of IndexDefinitions of unique indexes that apply to this class."""
+        return frozenset({
+            index_definition
+            for index_definition in self.get_all_indexes_for_class(cls)
+            if index_definition.unique
+        })
+
+    def get_properties_covered_by_index(self, index_definition, classname, props):
+        """Return the dict of values captured by the index, or None if the index does not apply.
+
```

Comment:
I would add the sentence `An index capturing properties means that it prevents these property values from appearing in two different entries`.

Reply:
Is this always true? What if the unique field is false?

Reply:
If we change `captured` to `covered`, we'd be using the standard nomenclature for this and we probably wouldn't need additional explanation.